packages/runtime/src/decorators/response.ts

```diff
@@ -1,10 +1,22 @@
-export function SuccessResponse(name: string | number, description?: string): Function {
+export function SuccessResponse(name: string | number, description?: string): Function;
+
+export function SuccessResponse<T>(name: string | number, description?: string): Function;
+
+export function SuccessResponse<T>(name: string | number, description?: string): Function {
   return () => {
     return;
   };
 }
 
-export function Response<T>(name: string | number, description?: string, example?: T): Function {
+export function Response(name: string | number, description?: string): Function;
+
+export function Response<U>(name: string | number, description?: string): Function;
+
+export function Response<T>(name: string | number, description?: string, example?: T): Function;
+
+export function Response<T, U>(name: string | number, description?: string, example?: T): Function;
+
+export function Response<T, U>(name: string | number, description?: string, example?: T): Function {
```

Comment:
1. Prefer more helpful names (Schema, Headers) over generic T, U
2. Add some restrictions for the Header type

Reply:
@WoH sorry, I don't quite understanding this statements.  
```ts
Header extends keyof Header extends string ? unknown : 'Header Type must only have string keys';
```
Can you provide some example, thank you!

Reply:
```ts
export type TsoaResponse<T extends HttpStatusCodeLiteral, U, Header extends keyof Header extends string ? unknown : 'Header Type must only have string keys';> = (status: T, data: U, headers?: V) => any;

```

That said, I think it makes sense to also restrict the values to `string | string[]`, having to deal with anything else is a headache.

Reply:
Applied this now. While your initial approach would probably work, I looked at the header typings and the consensus (tsoa, express) usually is that values must be strings, so I applied that.

Maybe we can loosen that w/ regards to numbers, i.e. fastify types headers as

```ts
  header(key: string, value: any): FastifyReply<RawServer, RawRequest, RawReply, RouteGeneric, ContextConfig>;
  headers(values: {[key: string]: any}): FastifyReply<RawServer, RawRequest, RawReply, RouteGeneric, ContextConfig>;
  getHeader(key: string): string | undefined;
  getHeaders(): {
    // Node's `getHeaders()` can return numbers and arrays, so they're included here as possible types.
    [key: string]: number | string | string[] | undefined;
  };
```

Since it's all a string after returning, I think a string with pattern probably is a more precise spec. Let me know what you think.